src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -87,37 +127,43 @@ public static class CreateMutationBuilder
         /// <param name="databaseType">The type of database to generate for</param>
         /// <param name="definitions">The other named types in the schema</param>
         /// <returns>true if the field is allowed, false if it is not.</returns>
-        private static bool FieldAllowedOnCreateInput(FieldDefinitionNode field, DatabaseType databaseType, IEnumerable<HotChocolate.Language.IHasName> definitions)
+        private static bool IsComplexFieldAllowedForCreateInput(FieldDefinitionNode field, DatabaseType databaseType, IEnumerable<HotChocolate.Language.IHasName> definitions)
         {
-            if (IsBuiltInType(field.Type))
-            {
-                // cosmosdb_nosql doesn't have the concept of "auto increment" for the ID field, nor does it have "auto generate"
-                // fields like timestap/etc. like SQL, so we're assuming that any built-in type will be user-settable
-                // during the create mutation
-                return databaseType switch
-                {
-                    DatabaseType.CosmosDB_NoSQL => true,
-                    _ => !IsAutoGeneratedField(field),
-                };
-            }
-
             if (QueryBuilder.IsPaginationType(field.Type.NamedType()))
             {
-                return false;
+                // Support for inserting nested entities with relationship cardinalities of 1-N or N-N is only supported for MsSql.
+                return databaseType is DatabaseType.MSSQL;
             }
 
             HotChocolate.Language.IHasName? definition = definitions.FirstOrDefault(d => d.Name.Value == field.Type.NamedType().Name.Value);
             // When creating, you don't need to provide the data for nested models, but you will for other nested types
             // For cosmos, allow updating nested objects
             if (definition != null && definition is ObjectTypeDefinitionNode objectType && IsModelType(objectType) && databaseType is not DatabaseType.CosmosDB_NoSQL)
             {
-                return false;
+                return databaseType is DatabaseType.MSSQL;
             }
 
             return true;
         }
 
-        private static InputValueDefinitionNode GenerateSimpleInputType(NameNode name, FieldDefinitionNode f)
+        private static bool IsBuiltInTypeFieldAllowedForCreateInput(FieldDefinitionNode field, DatabaseType databaseType)
+        {
+            // cosmosdb_nosql doesn't have the concept of "auto increment" for the ID field, nor does it have "auto generate"
+            // fields like timestap/etc. like SQL, so we're assuming that any built-in type will be user-settable
+            // during the create mutation
+            return databaseType switch
+            {
+                DatabaseType.CosmosDB_NoSQL => true,
+                _ => !IsAutoGeneratedField(field)
+            };
+        }
+
+        private static bool HasForeignKeyReference(FieldDefinitionNode field)
```

Comment:
nit: add method summary

Reply:
I think `IsAReferencingField` field makes more sense with its purpose. I'll rename it accordingly.